application/src/main/java/uk/gov/hmcts/dm/service/StoredDocumentService.java

```diff
@@ -156,11 +159,12 @@ public void deleteDocument(StoredDocument storedDocument, boolean permanent) {
         if (permanent) {
             storedDocument.setHardDeleted(true);
             storedDocument.getDocumentContentVersions().forEach(documentContentVersion -> {
-                Optional.ofNullable(documentContentVersion.getDocumentContent())
-                        .ifPresent(dc -> {
-                            documentContentRepository.delete(dc);
-                            documentContentVersion.setDocumentContent(null);
-                        });
+                if (azureStorageConfiguration.isAzureBlobStoreEnabled()) {
+                    blobStorageDeleteService.deleteDocumentContentVersion(documentContentVersion);
```

Comment:
It's not a change for now but in future it would be nice to wrap these two storage implementations behind a common interface so it can be hot swapped by the dependency injection container based on config